### PR TITLE
final

### PR DESCRIPTION
**Pull request by LitvinOleksandr**

### CODE REVIEW

File: src/main/java/core/basesyntax/Figure.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+public abstract class Figure implements CalculateArea, Draw {
+    private Color color;
+
+    public Figure(Color color) {
+        this.color = color;
+    }
+
+    public Figure() {
+
+    }
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/Figure.java:12]: do you really need this?

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    public Color getRandomColor() {
+        int index = new Random().nextInt(Color.values().length);
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: move random to field level

File: src/main/java/core/basesyntax/Main.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        //Write how many time would you like show result. Half iterations will be standard methods.
+        int iteration = 8;
+        Figure[] figures = new Figure[iteration];
+        for (int i = 0; i < iteration; i++) {
+            figures[i] = i < (iteration / 2) ? figureSupplier.getRandomFigure()
+                    : figureSupplier.getDefaultFigure();
+        }
+        for (Figure figure : figures) {
+            System.out.println(figure.writeText());
+        }
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/Main.java:15]: do it in one loop
create constant for iteration, think of better name

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+public class FigureSupplier {
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private static final Color DEFAULT_CIRCLE_COLOR = Color.WHITE;
+    private static final int VALUE_TO_RANDOM = 50;
+    private static final int FIGURE_AMMOUNT = 5;
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random randomNumber = new Random();
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/FigureSupplier.java:13]: move this to field level

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private static final Color DEFAULT_CIRCLE_COLOR = Color.WHITE;
+    private static final int VALUE_TO_RANDOM = 50;
```
Comment (dSkarzh) [inline src/main/java/core/basesyntax/FigureSupplier.java:8]: MAX_PROPERTY_VALUE

**Review by dSkarzh — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random randomNumber = new Random();
+        int randomFigure = randomNumber.nextInt(FIGURE_AMMOUNT);
+        switch (randomFigure) {
+            case 0: //Circle
+                int radius = 0;
+                while (radius == 0) {
+                    radius = randomNumber.nextInt(VALUE_TO_RANDOM);
+                }
```
[the remark this answers is not in the document]
Comment (LitvinOleksandr) [reply src/main/java/core/basesyntax/FigureSupplier.java:20]: radius can`t be a 0

**Pull request closed without merging**